vscode-dotnet-runtime-library/src/Acquisition/WinMacGlobalInstaller.ts

```diff
@@ -0,0 +1,304 @@
+                    reject(`Server responded with ${response.statusCode}: ${response.statusMessage}`);
+                }
+            });
+
+            request.on('error', err =>
+            {
+                file.close();
+                fs.unlink(dest, () => {}); // Delete incomplete file download
+                reject(err.message);
+            });
+
+            file.on('finish', () =>
+            {
+                resolve();
+            });
+
+            file.on('error', err =>
+            {
+                file.close();
+
+                if (err.message.includes('EEXIST'))
+                {
+                    // 2+ concurrent requests to download the installer occurred and ours got to the race last.
+                    resolve();
+                }
+                else
+                {
+                    fs.unlink(dest, () => {}); // Delete incomplete file download
+                    reject(err.message);
+                }
+            });
+        });
+    }
+
+    public async getExpectedGlobalSDKPath(specificSDKVersionInstalled : string, installedArch : string) : Promise<string>
+    {
+        if(os.platform() === 'win32')
+        {
+            // The program files should always be set, but in the off chance they are wiped, we can try to use the default as backup.
+            // Both ia32 and x64 machines will use 'Program Files'
+            // We don't anticipate a user would need to install the x86 SDK, and we don't have any routes that support that yet.
+            return path.resolve(path.join(process.env.programfiles!, 'dotnet', 'sdk') ?? `C:\\Program Files\\dotnet\\sdk\\`);
+        }
+        else if(os.platform() === 'darwin')
+        {
+            // On an arm machine we would install to /usr/local/share/dotnet/x64/dotnet/sdk` for a 64 bit sdk
+            // but we don't currently allow customizing the install architecture so that would never happen.
+            return path.resolve(`/usr/local/share/dotnet/sdk`);
+        }
+
+        const err = new DotnetUnexpectedInstallerOSError(new Error(`The operating system ${os.platform()} is unsupported.`));
+        this.acquisitionContext.eventStream.post(err);
+        throw err.error;
+    }
+
+    /**
+     *
+     * @param installerPath The path to the installer file to run.
+     * @returns The exit result from running the global install.
+     */
+    public async executeInstall(installerPath : string) : Promise<string>
+    {
+        this.commandRunner.returnStatus = true;
+        if(os.platform() === 'darwin')
+        {
+            // For Mac:
+            // We don't rely on the installer because it doesn't allow us to run without sudo, and we don't want to handle the user password.
+            // The -W flag makes it so we wait for the installer .pkg to exit, though we are unable to get the exit code.
+            const commandResult = await this.commandRunner.execute(`open -W ${path.resolve(installerPath)}`);
```

Comment:
`open` will not work if someone has f'd up their PATH on OSX. Please consider a fix for this or use the try alternative command function.

Reply:
fixed